### PR TITLE
Added code style linting and standardized the responses

### PR DESCRIPTION
**Pull request by silverAndroid**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
     res.json({
-        message: 'pong'
-    })
-})
-app.use('/users', require('./routes/user-routes'))
-app.use('/posts', require('./routes/post-routes'))
-app.use('/follow', require('./routes/follow-routes'))
-
-app.listen(process.env.PORT || 8080)
+        message: 'pong',
```
Comment (BrandonDanis) [inline index.js]: why is there a comma?

**Comment by BrandonDanis** [comment]
There is so much repeated code where you send back responses. Maybe we should create a `FSError` which inherits from the node `Error`, then whenever you want to return with a status code, you can throw an `FSError`, which can get caught by our catch all error handler, then it can do something like

```javascript
if (err instanceof FSError) {
      return res.status(err.status).send(this.serializer.serialize(err))
 }
next(err) //these are for unexpected errors 
```

**Comment by BrandonDanis** [comment]
It's about time we have code styling 👍

File: index.js (hunk 1 above)
Comment (jonathanGB) [reply index.js]: I like that, good habit :)

**Comment by rperera-bbm** [comment]
Yeah it would be a good idea to have some generic responses for less repetition

File: index.js (hunk 1 above)
Comment (rperera-bbm) [reply index.js]: Part of the coding style + if we want to add more properties, it'll show as an addition instead of a change

**Comment by silverAndroid** [comment]
Yeah it would be a good idea to add some generic methods for repeated responses

File: index.js (hunk 1 above)
Comment (silverAndroid) [reply index.js]: Part of the coding style + if we want to add more properties, it'll show as an addition instead of a change

File: auth.js [hunk 2]
```diff
@@ -13,28 +13,34 @@ module.exports.authenticate = async (req, res, next) => {
...
         } catch (e) {
+            console.error(e);
             switch (e.message) {
             case 'User not found':
                 return res.status(404).json({
-                    error: true,
-                    data: 'User not found',
+                    data: null,
+                    errors: [{ title: 'User not found' }],
+                    meta: {},
```
Comment (jonathanGB) [inline auth.js]: is `meta` used somewhere? @silverAndroid

File: docker-compose.yml [hunk 3]
```diff
@@ -24,3 +24,14 @@ services:
             - "5432:5432"
         volumes:
           - ./Database/db.sql:/docker-entrypoint-initdb.d/db.sql
+    friendstagram-db-test:
+        image: postgres:latest
+        environment:
+          # TODO: Use Docker secrets to hide this sensitive information
```
Comment (jonathanGB) [inline docker-compose.yml]: why not put docker secrets now?

Comment (jonathanGB) [inline docker-compose.yml]: why not put docker secrets now?

Comment (silverAndroid) [reply docker-compose.yml]: good point

File: auth.js (hunk 2 above)
Comment (silverAndroid) [reply auth.js]: I added it to follow this: http://jsonapi.org/format/#document-structure

Comment (silverAndroid) [reply auth.js]: but if we don't need this, I can remove it.
@BrandonDanis thoughts?

Comment (jonathanGB) [reply auth.js]: I'd say to only add it when necessary

Comment (BrandonDanis) [reply auth.js]: I agree, maybe only include if necessary

File: docker-compose.yml (hunk 3 above)
Comment (BrandonDanis) [reply docker-compose.yml]: all in one PR 👍🏽

**Comment by BrandonDanis** [comment]
Should have put the hacktoberfest tag on the issues and the PR 😉

**Comment by silverAndroid** [comment]
nothing stopping me from adding it to this 😜

**Comment by BrandonDanis** [comment]
@silverAndroid It needs to be on the issues too. You should have created one PR per issue, because now it's getting confusing with the amount of issues you're trying to resolve 😕

**Comment by silverAndroid** [comment]
I might just submit another PR with Docker secrets

**Comment by BrandonDanis** [comment]
I'll test this out later :)

**Comment by jonathanGB** [comment]
still believe that:
1. try {} catch makes it less readable
2. some logic could be simplified by early return

**Comment by BrandonDanis** [comment]
Okay so Travis now uses standard before tests. When developing, you can run `make test-fast` to just run component-tests, and you can run `make lint` to check syntax. CircleCI will run `make tests` which is both combined.

**Pull request merged**